AuthoringTests.md

```diff
@@ -4,55 +4,56 @@
+By default, the VFS for Git functional tests run a subset of tests as a quick smoke test for developers. To run all tests, pass in the `--full-suite` flag.
 
 ### 3. Running specific tests
 
-Specific tests can be run by specifying `--test=<comma separated list of tests>` as the command line arguments to the functional
-test project.
+Specific tests can be run by specifying `--test=<comma separated list of tests>` as the command line arguments to the functional test project.
 
 ### 4. How to write a functional test
 
-Each piece of functionality that we add to GVFS should have corresponding functional tests that clone a repo, mount GVFS, and use existing tools and file system
+Each piece of functionality that we add to VFS for Git should have corresponding functional tests that clone a repo, mount the filesystem, and use existing tools and filesystem
 APIs to interact with the virtual repo.
 
 Since these are functional tests that can potentially modify the state of files on disk, you need to be careful to make sure each test can run in a clean 
 environment.  There are three base classes that you can derive from when writing your tests.  It's also important to put your new class into the same namespace
 as the base class, because NUnit treats namespaces like test suites, and we have logic that keys off of that for deciding when to create enlistments.
 
-1. TestsWithLongRunningEnlistment
-    Before any test in this namespace is executed, we create a single enlistment and mount GVFS.  We then run all tests in this namespace that derive
+1. `TestsWithLongRunningEnlistment`
+
+    Before any test in this namespace is executed, we create a single enlistment and mount VFS for Git.  We then run all tests in this namespace that derive
 	from this base class.  Only put tests in here that are purely readonly and will leave the repo in a good state for future tests.
 
-2. TestsWithEnlistmentPerFixture
-    For any test fixture (a fixture is the same as a class in NUnit) that derives from this class, we create an enlistment and mount GVFS before running
+2. `TestsWithEnlistmentPerFixture`
+
+    For any test fixture (a fixture is the same as a class in NUnit) that derives from this class, we create an enlistment and mount VFS for Git before running
 	any of the tests in the fixture, and then we unmount and delete the enlistment after all tests are done (but before any other fixture runs).  If you need
 	to write a sequence of tests that manipulate the same repo, this is the right base class.
 
-3. TestsWithEnlistmentPerTestCase
-	Derive from this class if you need a brand new enlistment per test case.  This is the most reliable, but also most expensive option.
+3. `TestsWithEnlistmentPerTestCase`
+
+   Derive from this class if you need a brand new enlistment per test case.  This is the most reliable, but also most expensive option.

```

Comment:
One other thing that would should add here is the use of `GitRepoTests` and use it if you want to test a git command and have the results of a GVFS repo compared against a regular git clone repo.

Reply:
Unfortunately, the repo containing `GitRepoTests` is not public, right? (It contains specific branch and commit names for private, large repos.) So, that is something for our team wiki and not the public docs.

Reply:
The `GitRepoTests` is an `abstract` class in the functional tests that is used to run git commands and compare the output of the git command on GVFS and on a regular git clone.  [This is all public](https://github.com/Microsoft/VFSForGit/blob/655ba9de69843c3b64e20cecaaca2c43bb4466fa/GVFS/GVFS.FunctionalTests/Tests/GitCommands/GitRepoTests.cs#L14) All the tests that are in the `Tests/GitCommands` folder use this base class.

Reply:
Neat! Consider me corrected. I was not familiar with that.

Reply:
We should probably rename it to something like `TestWithGitControlRepo` to match the other base class names and avoid the confusion.